### PR TITLE
Expand TUF checkup

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Expands the current checkup to pull additional data.

Example output:

```
{
    "launcher_versions_in_library":[
        "/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2",
        "/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2-13-g2789a54",
        "/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2-15-gaa6f573"
    ],
    "local_metadata_version":246,
    "osqueryd_versions_in_library":[
        "/var/kolide-k2/k2device-preprod.kolide.com/updates/osqueryd/5.10.1"
    ],
    "remote_metadata_version":246,
    "selected_versions":{
        "launcher":{
            "path":"/Users/rebeccamahany-horton/Repos/launcher/build/darwin.universal/Kolide.app/Contents/MacOS/launcher",
            "version": ""
        },
        "osqueryd":{
            "path":"/var/kolide-k2/k2device-preprod.kolide.com/updates/osqueryd/5.10.1/osquery.app/Contents/MacOS/osqueryd",
            "version": "5.10.1"
        }
    }
}
```

### CODE REVIEW

**Review by zackattack01 — APPROVED** [review]
nice 🚀

File: pkg/debug/checkups/tuf.go [hunk 1]
```diff
@@ -54,51 +58,160 @@ func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
...
+		tufData["osqueryd_versions_in_library"] = fmt.Sprintf("error getting versions available in local osqueryd library: %v", err)
+	} else {
+		tufData["osqueryd_versions_in_library"] = localOsqueryVersions
+	}
+
+	tufData["selected_versions"] = tc.selectedVersions()
+
+	if b, err := json.Marshal(tufData); err == nil {
+		_, _ = extraFH.Write(b)
+	}
```
Comment (directionless) [inline pkg/debug/checkups/tuf.go:112]:
```suggestion
	if err := json.NewEncoder(extraFH).Encode(tufData); err != nil
		tc.summary += " cannot write extra data"
	}
```

maybe? Or drop the error handling from that.

File: pkg/debug/checkups/tuf.go [hunk 2]
```diff
@@ -54,51 +58,160 @@ func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
...
+
+	osquerydVersionMatchPattern := filepath.Join(updatesDir, "osqueryd", "*")
+	osquerydMatches, err := filepath.Glob(osquerydVersionMatchPattern)
+	if err != nil {
+		return nil, fmt.Errorf("globbing for osqueryd matches at %s: %w", osquerydVersionMatchPattern, err)
+	}
+
+	return osquerydMatches, nil
+}
```
Comment (directionless) [inline pkg/debug/checkups/tuf.go:192]: I'd probably merge these and take the match string.

File: pkg/debug/checkups/tuf.go [hunk 3]
```diff
@@ -54,51 +58,160 @@ func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
...
+
+// selectedVersions returns the versions of launcher and osqueryd that the current
+// installation would select as the correct version to run.
+func (tc *tufCheckup) selectedVersions() map[string]map[string]string {
```
Comment (directionless) [inline pkg/debug/checkups/tuf.go:196]: This is neat.

**Review by directionless — APPROVED** [review]
Small comments. Could merge as-is

File: pkg/debug/checkups/tuf.go (hunk 2 above)
Comment (RebeccaMahany) [reply pkg/debug/checkups/tuf.go:192]: The match string will be long/annoying to read in that case -- e.g. for me it'd be `"/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2,/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2-13-g2789a54,/var/kolide-k2/k2device-preprod.kolide.com/updates/launcher/1.1.2-15-gaa6f573"`. (With the array approach, whatever I use to read the JSON file will split it up onto separate lines and make it much more readable.) If you'd rather a string, I think I should iterate through and call `filepath.Base` on each so we end up with `"1.1.2,1.1.2-13-g2789a54,1.1.2-15-gaa6f573"`. Which do you prefer?

Comment (RebeccaMahany) [reply pkg/debug/checkups/tuf.go:192]: (I'd been kind of wanting to do `filepath.Base` anyway so I'm happy with that plus a join!)

Comment (directionless) [reply pkg/debug/checkups/tuf.go:192]: I don't feel strongly. I noticed the duplicated functions that look like they differ only by a the `"launcher"` or `"osqueryd"` strings.

Comment (RebeccaMahany) [reply pkg/debug/checkups/tuf.go]: _oh_

Comment (RebeccaMahany) [reply pkg/debug/checkups/tuf.go]: Sorry, I keep missing which lines a comment applies to on the diff in GitHub -- updated to combine

Comment (directionless) [reply pkg/debug/checkups/tuf.go]: GitHub can be _something_

**Review by directionless — APPROVED** [review]

**Pull request merged**